presto-hive/src/test/java/com/facebook/presto/hive/orc/TestOrcBatchPageSourceFactory.java

```diff
@@ -0,0 +1,103 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.orc;
+
+import com.facebook.presto.common.predicate.TupleDomain;
+import com.facebook.presto.hive.DwrfEncryptionMetadata;
+import com.facebook.presto.hive.EncryptionInformation;
+import com.facebook.presto.hive.FileFormatDataSourceStats;
+import com.facebook.presto.hive.HiveColumnHandle;
+import com.facebook.presto.hive.HiveDwrfEncryptionProvider;
+import com.facebook.presto.hive.HiveFileSplit;
+import com.facebook.presto.hive.TestOrcBatchPageSourceMemoryTracking;
+import com.facebook.presto.orc.OrcReaderOptions;
+import com.facebook.presto.orc.StorageStripeMetadataSource;
+import com.facebook.presto.orc.StripeMetadataSourceFactory;
+import com.facebook.presto.orc.cache.StorageOrcFileTailSource;
+import com.facebook.presto.spi.ConnectorPageSource;
+import com.facebook.presto.spi.PrestoException;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.ql.io.orc.OrcOutputFormat;
+import org.apache.hadoop.hive.ql.io.orc.OrcSerde;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.joda.time.DateTimeZone;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+import static com.facebook.presto.hive.HiveFileContext.DEFAULT_HIVE_FILE_CONTEXT;
+import static com.facebook.presto.hive.HiveTestUtils.FUNCTION_AND_TYPE_MANAGER;
+import static com.facebook.presto.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static com.facebook.presto.hive.HiveTestUtils.SESSION;
+import static com.facebook.presto.orc.OrcEncoding.ORC;
+import static org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory.javaStringObjectInspector;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class TestOrcBatchPageSourceFactory
+{
+    @Test
+    public void testCreateOrcPageSourceException() throws Exception
+    {
+        Random random = new Random();
+        List<TestOrcBatchPageSourceMemoryTracking.TestColumn> testColumns = ImmutableList.<TestOrcBatchPageSourceMemoryTracking.TestColumn>builder()
+                .add(new TestOrcBatchPageSourceMemoryTracking.TestColumn("p_empty_string", javaStringObjectInspector, () -> "", true))
+                .add(new TestOrcBatchPageSourceMemoryTracking.TestColumn("p_string", javaStringObjectInspector, () -> Long.toHexString(random.nextLong()), false))
+                .build();
+        File tempFile = File.createTempFile("presto_test_orc_batch_page_source_factory", "orc");
+        tempFile.delete();
```

Comment:
You create the file and immediately delete it? This is strange

Reply:
I used the same setup for the test https://github.com/prestodb/presto/blob/be299743bf1b2feaaf0fbfaa3d09ce052908a22b/presto-hive/src/test/java/com/facebook/presto/hive/TestOrcBatchPageSourceMemoryTracking.java#L168  - looking at the code the idea is to get the absolute path of the temporary file as it is later used by a writer to write content into that  path after which a file is created.

Reply:
That feels rather roundabout. I suspect there's a simple and more reliable way to do this without creating and deleting the file. Just use the path and System.getProperty("java.io.tmpdir") for instance.